collections/zip.ts

```diff
@@ -22,15 +22,15 @@
  * ```
  */
 export function zip<T, U>(
-  array: Array<T>,
-  withArray: Array<U>,
-): Array<[T, U]> {
+  array: readonly T[],
+  withArray: readonly U[],
+): [T, U][] {
   const returnLength = Math.min(
     array.length,
     withArray.length,
   );
 
-  const ret: Array<[T, U]> = [];
+  const ret: [T, U][] = [];
```

Comment:
We should use the calculated `returnLength` here (I missed that on my initial implementation):

```suggestion
  const ret = new Array<[T, U]>(returnLength);
```